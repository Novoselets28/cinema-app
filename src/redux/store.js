import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer from './reducers/reducersSessionTime';
import sessionListReducer from './reducers/reducersSessionList';
import sessionTimeSaga from './sagas/sagasSessionTime';
import sessionListSaga from './sagas/sagasSessionList';
import movieSessionsReducer from './reducers/reducersMovieSessions';
import movieSessionsSaga from './sagas/sagasMovieSessions';
import mainPageReducer from './reducers/reducersMainPage';
import mainPageSaga from './sagas/sagasMainPage';
import filmDescriptionReducer from './reducers/filmDescriptionReducer';
import filmDescriptionSaga from './sagas/filmDescriptionSaga';
import cinemaReducer from './reducers/reducersCinema';
import cinemaSaga from './sagas/cinemaSaga';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,

});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;
