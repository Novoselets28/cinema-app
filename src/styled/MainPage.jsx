import { Container, Row } from 'react-bootstrap';
import styled from 'styled-components';


export const StyledContainer = styled(Container)`
    max-width: 100%;
    text-align: center;
    height: 500px;
`;

export const StyledRow = styled(Row)`
    display: flex;
    justify-content: center;
`;