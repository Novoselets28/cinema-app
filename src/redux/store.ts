import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer, { sessionTimeSaga } from './ducks/sessionTime';
import filmDescriptionReducer, { filmDescriptionSaga } from './ducks/filmDescription';
import cinemaReducer, { cinemaSaga } from './ducks/cinema';
import sessionListReducer, { sessionListSaga } from './ducks/sessionList';
import movieSessionsReducer, { movieSessionsSaga } from './ducks/movieSessions';
import mainPageReducer, { mainPageSaga } from './ducks/mainPage';

export type RootState = ReturnType<typeof rootReducer>;

const sagaMiddleware = createSagaMiddleware();

export const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,
});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;