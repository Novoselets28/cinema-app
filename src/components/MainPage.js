import React, { useState, useEffect } from 'react';
import { Carousel, Col } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { StyledContainer, StyledRow } from '../styled/MainPage';
import MovieSessions from './MovieSessions';
import {API_URL_LIST_OF_FILM} from '../api';

const MainPage = () => {
  const [posters, setPosters] = useState([]);

  useEffect(() => {
    fetch(API_URL_LIST_OF_FILM)
      .then((response) => response.json())
      .then((data) => {
        const postersArray = data.list.map((film) => film.Poster);
        setPosters(postersArray);
      })
      .catch((error) => console.error('Error fetching posters:', error));
  }, []);

  return (
    <StyledContainer fluid>
        <MovieSessions/>
        <StyledRow>
        <Col xs={12} md={6} lg={4} xxl={3}>
            <Carousel>
            {posters.map((poster, index) => (
            <Carousel.Item key={index}>
                <img className="d-block w-100" src={poster} alt={`Poster ${index + 1}`} />
            </Carousel.Item>
            ))}
        </Carousel>
        </Col>
    </StyledRow>
    </StyledContainer>
    
        
  );
};

export default MainPage;
