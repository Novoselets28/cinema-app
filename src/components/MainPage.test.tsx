import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux'; // Import Provider
import MainPage from './MainPage';
import store from '../redux/store';
import { MemoryRouter } from 'react-router';


test('MainPage renders correctly', () => {
  // Wrap the component with the Redux Provider and provide the mock store
  const { container } = render(
    <Provider store={store}>
        <MemoryRouter>
            <MainPage mainPage={''} selectedSession={''} />
        </MemoryRouter>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
