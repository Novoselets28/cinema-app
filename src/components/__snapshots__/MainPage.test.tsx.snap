// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<div>
  <div
    class="sc-aXZVg iSVraJ container-fluid"
  >
    <div
      class="sc-fqkvVR jWBLYB movie-sessions-page container"
    >
      <div
        class="card-header"
      >
        Available dates for booking tickets
      </div>
    </div>
    <div
      class="sc-gEvEer jytxZI row"
    >
      <div
        class="col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="carousel slide"
        >
          <div
            class="carousel-indicators"
          />
          <div
            class="carousel-inner"
          />
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
